server/chat-commands/core.ts

```diff
@@ -732,10 +783,14 @@ export const commands: ChatCommands = {
 			}
 			for (const setting in user.settings) {
 				if (setting in raw) {
-					if (setting === 'blockPMs' &&
-						Users.Auth.isAuthLevel(raw[setting])) {
-						settings[setting] = raw[setting];
+					if (setting === 'blockPMs') {
+						const {all, specific} = raw[setting];
+						settings.blockPMs = {
+							all: typeof all === 'boolean' || Users.Auth.isAuthLevel(all) ? all : false,
+							specific: specific && Array.isArray(specific) ? specific : [],
+						};
 					} else {
+						// @ts-ignore (the only exception is parsed above, so this can be ignored)
```

Comment:
What is the error here? You should not need to ignore anything

Reply:
> Type 'boolean' is not assignable to type '(false & { all: boolean | "!" | "~" | "&" | "#" | "★" | "*" | "@" | "%" | "☆" | "+" | " " | "‽" | "whitelist" | "unlocked" | "trusted" | "autoconfirmed"; specific?: string[] | undefined; }) | (true & { ...; }) | undefined'.

Reply:
it was previously fine with the typeof .... === object fix, but that's off the table

Reply:
use `as keyof Omit<UserSettings, 'blockPMs'>`: https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys